api/src/wfl/executor.clj

```diff
@@ -428,6 +446,91 @@
      executor
      (jdbc/query tx [(format query details) status]))))
 
+(defn ^:private reference-to-workflow-records
+  "Find all submission IDs associated with `workflows`.
+  Return a mapping of the submission reference (ex. snapshot reference id)
+  to their associated workflow records."
+  [tx {:keys [details] :as executor} workflows]
+  (throw-if-no-details-table tx executor)
+  (let [workflow-ids (util/to-quoted-comma-separated-list (map :uuid workflows))
+        _            (-> "%s Fetching submission IDs linked to workflows %s"
+                         (format (log-prefix executor) workflow-ids)
+                         log/debug)
+        ;; Presently Rawls doesn't support submitting a subset of a snapshot:
+        ;; If we wish to retry 1+ workflows stemming from a snapshot,
+        ;; we must resubmit the snapshot in its entirety.
+        query "SELECT * FROM %s
+               WHERE submission IN
+                 (SELECT DISTINCT submission FROM %s
+                 WHERE workflow IN %s)"]
```

Comment:
Don't you want distinct references rather than submissions? Seems like the logic is a bit twisted?

Reply:
I flip-flopped between distinct references and submissions, agreed that it is twisty.

My thoughts were that…
- When a reference is submitted successfully it yields one submission
- One submission maps to one set of workflows
- Given the capacity for retries, one reference can map to many sets of workflows

That Rawls limitation on submitting snapshot subsets again makes things hairy here.  I want to get all of the workflow records for the same submissions as the ones we need to retry, since they'll all be retried.

A possible alternative, though I couldn't convince myself 100% of its infallibility it may make more sense to everyone else:
```
"SELECT * FROM <details>
 WHERE retry IS NULL
 AND reference IN 
     (SELECT DISTINCT reference FROM %s
     WHERE workflow IN %s)

 ```

Reply:
I will add a comment to this effect and file a ticket to make this threadsafe.

Reply:
Comment added, and ticket: https://broadinstitute.atlassian.net/browse/GH-1423